test/e2e/utils_test.go

```diff
@@ -204,6 +205,77 @@ func verifyDeploymentArgs(k8sclient *kubernetes.Clientset, deploymentName string
 	})
 }
 
+// addOverrideResources adds the override resources to the specific cert-manager operand. The update process
+// is retried if a conflict error is encountered.
+func addOverrideResources(client *certmanoperatorclient.Clientset, deploymentName string, res v1alpha1.CertManagerResourceRequirements) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		operator, err := client.OperatorV1alpha1().CertManagers().Get(context.TODO(), "cluster", v1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		updatedOperator := operator.DeepCopy()
+
+		switch deploymentName {
+		case certmanagerControllerDeployment:
+			updatedOperator.Spec.ControllerConfig = &v1alpha1.DeploymentConfig{
+				OverrideResources: res,
+			}
+		case certmanagerWebhookDeployment:
+			updatedOperator.Spec.WebhookConfig = &v1alpha1.DeploymentConfig{
+				OverrideResources: res,
+			}
+		case certmanagerCAinjectorDeployment:
+			updatedOperator.Spec.CAInjectorConfig = &v1alpha1.DeploymentConfig{
+				OverrideResources: res,
+			}
+		default:
+			return fmt.Errorf("unsupported deployment name: %s", deploymentName)
+		}
+
```

Comment:
In the future this can cause a problem if we construct a complex test case (for some generic use case) that combines both `overrideArgs` and `overrideResources`. At that time we can check if `*Config` is nil / empty before proceeding to update with `&v1alpha1.DeploymentConfig{...}` directly.
But for now it should be fine, should block this PR (this is just me noting to myself :).